### PR TITLE
Allow EXP Records factory to generate automatically-awarded records

### PR DESCRIPTION
**Pull request by kxmbrian**

### CODE REVIEW

File: spec/factories/course_experience_points_records.rb [hunk 1]
```diff
@@ -11,10 +11,15 @@
         build(:course_user, course: course, user: user)
     end
     points_awarded { rand(1..20) * 100 }
-    reason 'EXP for some event'
+    actable_type 'self::PointsAwardingDummy'
```
Comment (lowjoel) [inline spec/factories/course_experience_points_records.rb]: This isn't a good idea. The polymorphic column is now in an indeterminate state. The type is specified but not the id

**Comment by lowjoel** [comment]
I'm not sure if this is a bad rebase but the spec change seems to be part of this PR. Is it deliberate? I'd prefer them to at least be in separate commits.

File: spec/factories/course_experience_points_records.rb (hunk 1 above)
Comment (kxmbrian) [inline spec/factories/course_experience_points_records.rb]: Okay, so I should create a dummy model with its database table?

Comment (lowjoel) [inline spec/factories/course_experience_points_records.rb]: A dummy model defined in the factory would be good.

In spec/factories/course_experience_points_records.rb<https://github.com/Coursemology/coursemology2/pull/909#discussion_r55920185>:

> @@ -11,10 +11,15 @@
>          build(:course_user, course: course, user: user)
>      end
>      points_awarded { rand(1..20) * 100 }
> -    reason 'EXP for some event'
> +    actable_type 'self::PointsAwardingDummy'


Okay, so I should create a dummy model with its database table?

File: spec/factories/course_experience_points_records.rb [hunk 2]
```diff
@@ -11,7 +11,17 @@
         build(:course_user, course: course, user: user)
     end
     points_awarded { rand(1..20) * 100 }
-    reason 'EXP for some event'
+
+    after(:build) do |record|
+      record.reason = 'Reason for manually-awarded experience points' if record.manually_awarded?
```
Comment (lowjoel) [inline spec/factories/course_experience_points_records.rb]: is this necessary?

Comment (kxmbrian) [inline spec/factories/course_experience_points_records.rb]: The model currently validates that there is a reason if the manually_awarded? is true for the record.

Comment (lowjoel) [inline spec/factories/course_experience_points_records.rb]: this should be an attribute in a block. `reason { 'REason...' if manually_awarded? }`

Comment (kxmbrian) [inline spec/factories/course_experience_points_records.rb]: okay, fixed.

**Pull request merged**